encoding/protobuf/error.go

```diff
@@ -170,25 +170,29 @@ func createStatusWithDetail(pberr *pberror, encoding transport.Encoding, codec *
 }
 
 func setApplicationErrorMeta(pberr *pberror, resw transport.ResponseWriter) {
-	applicationErroMetaSetter, ok := resw.(transport.ApplicationErrorMetaSetter)
+	applicationErrorMetaSetter, ok := resw.(transport.ApplicationErrorMetaSetter)
 	if !ok {
 		return
 	}
 
-	decodedDetails := GetErrorDetails(pberr)
-	var appErrName string
-	if len(decodedDetails) > 0 { // only grab the first name since this will be emitted with metrics
-		appErrName = messageNameWithoutPackage(proto.MessageName(
-			decodedDetails[0].(proto.Message)),
-		)
-	}
+	var (
+		decodedDetails = GetErrorDetails(pberr)
+
+		appErrName = ""
```

Comment:
nit: you can keep `var appErrName` without initializing. The zero value is already "". Can we just do `appErrName string` here?